src/date-picker/src/panel/use-panel-common.ts

```diff
@@ -32,6 +33,7 @@ const usePanelCommonProps = {
   },
   shortcuts: Object as PropType<Shortcuts>,
   defaultTime: [Number, String, Array] as PropType<DefaultTime>,
+  onClick: Function,
```

Comment:
not sure if this is needed, click should be a native event. 🤔

Reply:
> not sure if this is needed, click should be a native event. 🤔

I'm not quite sure how to add native events. I tried using native-on-click='event', but it didn't work. Do you have any suggestions

Reply:
like this :

```
  <MonthPanel
    ref="monthPanelRef"
    onUpdateValue={this.onUpdateValue}
    actions={[]}
    // month and year click show month type
    type="month"
    key="month"
    useAsQuickJump
    value={this.value}
    native-on-click={
      this.handleClosePanelOnSelectDate
    }
  />
```

Reply:
does it work if you omit `onClick: Function`?

Reply:
it is worked,  but i got a type error, maybe need the click prop？

```
error TS2322: Type '{ ref: string; onUpdateValue: (value: number) => void; actions: never[]; type: "month"; key: string; useAsQuickJump: true; value: number | undefined; onClick: () => void; }' is not assignable to type 'IntrinsicAttributes & Partial<{ value: Value | null; active: boolean; actions: string[]; closePanelOnSelectDate: boolean; panel: boolean; useAsQuickJump: boolean; }> & Omit<...>'.
  Property 'onClick' does not exist on type 'IntrinsicAttributes & Partial<{ value: Value | null; active: boolean; actions: string[]; closePanelOnSelectDate: boolean; panel: boolean; useAsQuickJump: boolean; }> & Omit<...>'.
```

Reply:
ill have a deeper look tomorrow, but maybe you need to omit this line too (https://github.com/tusen-ai/naive-ui/pull/5564/files#diff-bfa663dd0eb5b256856ff25045eb702e4904cf4dbb07b92866b236ac9bd359abR142)

Reply:
onclick on monthpanel should work native without listening to click on div explicitly

Reply:
some related questions， maybe there's some reference,  https://github.com/vuejs/babel-plugin-jsx/issues/555

Reply:
any progress or areas for improvement? I will respond promptly. tks. 😄

Reply:
> any progress or areas for improvement? I will respond promptly. tks. 😄

Not from my side. I am only a "first level" reviewer and do not decide whether this is sufficient or can be merged. I would simply include @07akioni  in the discussion here.

Reply:
@OrbisK If possible I'd prefer declared props. Native events has some implicit behavior.

Explicit behavior would make intentions easier to be noticed. Also let you know there's little possibility to make things wrong.